src/main/java/fr/techad/edc/httpd/search/ContentSearcher.java

```diff
@@ -89,12 +93,37 @@ public List<DocumentationSearchResult> search(String search, String lang, int li
       results.add(documentationSearchResult);
 
     }
-    if (results.isEmpty() && !defaultLanguage.equals(lang)) {
+    if (results.isEmpty() && !defaultLanguage.equals(lang) && checkLanguageIsPresent(lang) == false) {
       return search(search, defaultLanguage, limit, exact, defaultLanguage);
     }
     return results;
   }
 
+  private boolean checkLanguageIsPresent(String lang) throws IOException {
```

Comment:
This method should not check from a file. The goal of this class is to search. the list of languages have to be an argument of the constructor. The structure of the edc export should be out this class

Reply:
Don't use LangUtil here. The list of language have to be added as paramters in the search method.